tensorflow/core/platform/s3/s3_file_system.cc

```diff
@@ -217,9 +231,13 @@ class S3RandomAccessFile : public RandomAccessFile {
     });
     auto getObjectOutcome = this->s3_client_->GetObject(getObjectRequest);
     if (!getObjectOutcome.IsSuccess()) {
-      n = 0;
-      *result = StringPiece(scratch, n);
-      return Status(error::OUT_OF_RANGE, "Read less bytes than requested");
+      auto error = getObjectOutcome.GetError();
+      if (error.GetResponseCode() == Aws::Http::HttpResponseCode::REQUESTED_RANGE_NOT_SATISFIABLE) {
```

Comment:
Merged a fix for a bug when reading files, which treats error as end of file. This was earlier a separate PR: https://github.com/tensorflow/tensorflow/pull/36389 . Merging it here to save some time and effort into getting another PR in.